openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -276,6 +280,13 @@ def run(self, problem):
             # Use pyOpt's internal finite difference
             fd_step = problem.root.fd_options['step_size']
             sol = opt(opt_prob, sens='FD', sensStep=fd_step, storeHistory=self.hist_file)
+        elif self.options['snopt_diff']:
+            if self.options['optimizer']=='SNOPT':
+                fd_step = problem.root.fd_options['step_size']
+                sol = opt(opt_prob, sens=None, sensStep=fd_step, storeHistory=self.hist_file)
+            else:
+                msg = "SNOPT's internal finite difference method can only be used with SNOPT"
+                raise Exception(msg)
```

Comment:
We now have 3 mutually exclusive possibilities in terms of selecting the gradient method.  I suggest that instead of having two different flags (pyopt_diff and snopt_diff), We use a single enum, maybe 'gradient_method', with 3 values, maybe 'pyopt_fd', 'snopt_fd', and 'openmdao', with 'openmdao' being the default.  The 'add_option' command has a 'values' arg that lets you set enumerated values.  If we keep two separate flags, we would need to add error checking to make sure that both flags were not set at the same time.  Currently if pyopt_diff is set, snopt_diff is ignored.